FreydCategoriesForCAP/gap/precompiled_categories/CategoryOfColumnsAsOppositeOfCategoryOfRowsOfHomalgExteriorRingOverFieldPrecompiled.gi

```diff
@@ -126,15 +126,26 @@ end
         
 ########
 function ( cat_1, source_1, alpha_1, beta_1, range_1 )
-    local deduped_2_1, deduped_3_1, deduped_4_1, deduped_5_1, deduped_6_1;
-    deduped_6_1 := UnderlyingMatrix( alpha_1 );
-    deduped_5_1 := BasisOfRingOverBaseFieldAsColumnVector( cat_1 );
-    deduped_4_1 := UnderlyingMatrix( beta_1 );
-    deduped_3_1 := UnderlyingRing( cat_1 );
-    deduped_2_1 := RangeCategoryOfHomomorphismStructure( cat_1 );
-    return CreateCapCategoryMorphismWithAttributes( deduped_2_1, source_1, range_1, UnderlyingMatrix, CoercedMatrix( deduped_3_1, UnderlyingRing( deduped_2_1 ), CoefficientsWithGivenMonomials( KroneckerMat( TransposedMatrix( deduped_4_1 ), DualKroneckerMat( UnionOfRows( deduped_3_1, NumberColumns( deduped_5_1 ), List( EntriesOfHomalgColumnVector( deduped_5_1 ), function ( x_2 )
-                        return COMPILATION_HELPER_HomalgMatrixFromRingElement( x_2, deduped_3_1 );
-                    end ) ), deduped_6_1 ) ), DiagMat( deduped_3_1, ListWithIdenticalEntries( NumberRows( deduped_4_1 ), DiagMat( deduped_3_1, ListWithIdenticalEntries( NumberColumns( deduped_6_1 ), deduped_5_1 ) ) ) ) ) ) );
+    local hoisted_2_1, hoisted_4_1, hoisted_5_1, hoisted_6_1, deduped_7_1, deduped_8_1, deduped_9_1, deduped_10_1, deduped_11_1, deduped_12_1, deduped_13_1, deduped_14_1;
+    deduped_14_1 := BasisOfRingOverBaseFieldAsColumnVector( cat_1 );
+    deduped_13_1 := UnderlyingMatrix( alpha_1 );
+    deduped_12_1 := UnderlyingMatrix( beta_1 );
+    deduped_11_1 := UnderlyingRing( cat_1 );
+    deduped_10_1 := RangeCategoryOfHomomorphismStructure( cat_1 );
+    deduped_9_1 := EntriesOfHomalgColumnVector( deduped_14_1 );
+    deduped_8_1 := NumberRows( deduped_12_1 );
+    deduped_7_1 := DiagMat( deduped_11_1, ListWithIdenticalEntries( deduped_8_1, DiagMat( deduped_11_1, ListWithIdenticalEntries( NumberColumns( deduped_13_1 ), deduped_14_1 ) ) ) );
+    hoisted_6_1 := Sum( ListWithIdenticalEntries( NumberRows( deduped_13_1 ), Length( deduped_9_1 ) ) );
+    hoisted_5_1 := [ 1 .. deduped_8_1 ];
+    hoisted_4_1 := DualKroneckerMat( UnionOfRows( deduped_11_1, NumberColumns( deduped_14_1 ), List( deduped_9_1, function ( x_2 )
+                return COMPILATION_HELPER_HomalgMatrixFromRingElement( x_2, deduped_11_1 );
+            end ) ), deduped_13_1 );
+    hoisted_2_1 := OneImmutable( deduped_11_1 );
+    return CreateCapCategoryMorphismWithAttributes( deduped_10_1, source_1, range_1, UnderlyingMatrix, CoercedMatrix( deduped_11_1, UnderlyingRing( deduped_10_1 ), CoefficientsWithGivenMonomials( UnionOfRows( deduped_11_1, NumberColumns( deduped_7_1 ), List( [ 1 .. NumberColumns( deduped_12_1 ) ], function ( j_2 )
+                    return UnionOfColumns( deduped_11_1, hoisted_6_1, List( hoisted_5_1, function ( i_3 )
+                              return hoisted_2_1 * \[\,\]( deduped_12_1, i_3, j_2 ) * hoisted_4_1;
```

Comment:
I think the multiplication `hoisted_2_1 * ...` with `hoisted_2_1 := OneImmutable( deduped_11_1 )` is the reason for the changed code.

Reply:
Do you see a way to avoid this?

Reply:
> Do you see a way to avoid this?

Of course: A logic template dropping multiplication with `OneImmutable` for homalg ring elements.

Reply:
Ah, my brain excluded the option logic templates, I believe I know why :D